ateam_common/src/game_controller_listener.cpp

```diff
@@ -86,13 +92,46 @@ void TeamInfoListener::RefereeMessageCallback(
   if (team_color_ == TeamColor::Unknown) {
     team_side_ = TeamSide::Unknown;
   } else {
-    bool are_positive_half = !(msg->blue_team_on_positive_half ^ team_color_ == TeamColor::Blue);
+    bool are_positive_half = !(msg->blue_team_on_positive_half ^ (team_color_ == TeamColor::Blue));
     team_side_ = are_positive_half ? TeamSide::PositiveHalf : TeamSide::NegativeHalf;
     // Our field convention is we should always been on the negative half.
     // So if this is positive for our team we should invert coords
   }
   if (team_side_ != prev_side && side_callback_) {
     side_callback_(team_side_);
   }
+
+  game_stage_ = static_cast<GameStage>(msg->stage);
+
+  // Yellow commands are even, blue are odd
+  // we_blue  odd(blue command)
+  // 0        0                   ours    leave it
+  // 0        1                   theirs  leave it
+  // 1        0                   theirs  add by 1 to be theirs
+  // 1        1                   ours    sub by 1 to be ours
+
+
+  // ignore if we dont know our team color yet
+  // only force start and above have this mapping
+  uint8_t command = msg->command;
+  if (team_color_ != TeamColor::Unknown &&
+    command > static_cast<std::underlying_type_t<GameCommand>>(GameCommand::ForceStart))
+  {
+    if (team_color_ == TeamColor::Blue) {
+      bool command_is_blue = command % 2 == 1;
+      command += (command_is_blue ? -1 : 1);
+    }
+  }
```

Comment:
While clever, this feels brittle to changes in the command codes and feels less maintainable than simply checking the commands against the named constants and handling them.

Reply:
I replaced this with a switch statement that should have the same effect (translating the format from `ssl_league_msgs` into our `ateam_common` `GameCommand` type). Can you double check that the logic looks okay and that the code is more understandable?